### PR TITLE
Incense bugfixing

### PR DESCRIPTION
**Pull request by davidakachaos**
## Short Description:
Trying my first at catching incense Pokemon. Rebased on current dev.

## Fixes/Resolves/Closes (please use correct syntax):
- Catch Pokemon attracted by incense
- Check if an incense is active on a player
- Updated documentation for this change

Please review my changes and tell me if I broke something

### CODE REVIEW

File: pokemongo_bot/cell_workers/use_incense.py [hunk 1]
```diff
@@ -12,14 +13,23 @@ def initialize(self):
...
+
       self.types = {
         401: "Ordinary",
         402: "Spicy",
         403: "Cool",
         404: "Floral"
       }
-      
+
+    def _have_applied_incense(self):
```
Comment (sohje) [inline pokemongo_bot/cell_workers/use_incense.py]: Duplicate code from catch_pokemon.py ?

File: pokemongo_bot/cell_workers/use_incense.py [hunk 2]
```diff
@@ -30,35 +40,36 @@ def _get_type(self):
...
+
+      self._update_inventory()
+      if self._has_count() and self.use_incense:
+        return True
 
     def work(self):
       if self._should_run():
         self.start_time = time.time()
-        type = self._get_type()        
+        type = self._get_type()
```
Comment (sohje) [inline pokemongo_bot/cell_workers/use_incense.py]: type is a built-in func. Really bad idead to assign something to it.

File: pokemongo_bot/inventory.py [hunk 3]
```diff
@@ -319,6 +319,85 @@ def has_space_for_loot(cls):
...
+                ret = item
+                self.parse(item)
+
+        return ret
+
+    def __str__(self):
+        return self.name
+
+
+class AppliedItems(_BaseInventoryComponent):
```
Comment (sohje) [inline pokemongo_bot/inventory.py]: There is already:
```python
class Items(_BaseInventoryComponent):
    TYPE = 'item'
    ID_FIELD = 'item_id'
    STATIC_DATA_FILE = os.path.join(_base_dir, 'data', 'items.json')
```
Why did you add another 2 classes? Just extend items class. Also there is class Item(object)

File: pokemongo_bot/inventory.py [hunk 4]
```diff
@@ -1415,6 +1499,13 @@ def items():
     """
     return _inventory.items
 
+def applied_items():
```
Comment (sohje) [inline pokemongo_bot/inventory.py]: There is cached a _inventory.items object.

**Comment by davidakachaos** [comment]
Thanks for all the feedback @sohje 
I will have another go at it!

File: pokemongo_bot/inventory.py (hunk 3 above)
Comment (davidakachaos) [inline pokemongo_bot/inventory.py]: AppliedItem(s) is a seperate class in pgoapi, it are items that are in use by the player with a countdown. Like a LuckyEgg and a Incense. 
So it isn't the same as a incense in your inventory, it's a item that is being used.
my knowledge of Python is minimal (I learn as I go) so I just created two classes with Item and Items as example.

File: pokemongo_bot/inventory.py (hunk 4 above)
Comment (davidakachaos) [inline pokemongo_bot/inventory.py]: Yes, but here I'm not interested in the items in the bag. I want to get the items (and time left on those items) that are in use. Am I doing it wrong?

File: pokemongo_bot/cell_workers/use_incense.py (hunk 1 above)
Comment (davidakachaos) [inline pokemongo_bot/cell_workers/use_incense.py]: Yes, how would I dry this up?

**Comment by solderzzc** [comment]
👍

**Comment by javajohnHub** [comment]
Cant wait to try this out
